### ISSUE TITLE
Huge disk usage

### ISSUE THREAD

**Issue by ktrls** [Thomas101/wmail#663]
* Which version of WMail are you using? [e.g. 1.3.1]
wmail 2.3.1 prerelease

* Which Operating System are you using? [e.g. Windows 10, Mac OSX 10, Ubuntu 14.04]
Linux Manjaro with kernel 4.9.13-1-Manjaro

Hi, I've searched but it seems there is no issue about it.

I have inserted 8 gmail account with drive, contacts, etc and sometime my system read/write on the disk (maybe for swapping) freezing for almost a minut.

I tried to close applications in random order and then I can say that is Wmail the reason of this beavior.

What do you think about it?
There is some information that I can produce to help you and me for verification?

Thanks a lot

**Comment by jeroen7s** [comment]
how much ram do you have? since wmail uses a large ammount of ram it's likely that it starts swapping because ram is full
it basically consumes about 250MB per email account(at least for inbox, not sure about regular gmail) + 250MB for the base application, which would be around 2.25 GB of ram for 8 email accounts, can be a huge issue on 4gb ram, and even on 8gb ram it can be an issue with all the other applications you might have running at the same time

**Comment by jeroen7s** [comment]
there really should be an option to suspend the tabs so it can save a ton of ram. and to unsuspend it when you go to that particular email address, and poll every few minutes for updates instead of constantly keeping the site running to check for updates constantly

**Comment by ktrls** [comment]
Hi jeroen7s, I have 8 GB

**Comment by Thomas101** [comment]
Hi, do you see a lot of throughput with new emails etc? If you take a look in `~/.local/share/wmail/` whats the folder size? There should also be a few files with there with the extension .wmaildb, what's the filesize of those?

**Comment by ktrls** [comment]
This is my ll:

>[ktrls@pohl ~]$ ll ~/.local/share/wmail/
>-rw-r--r-- 1 ktrls ktrls    167 15 mar 09.36 app_db.json
>-rw-r--r-- 1 ktrls ktrls  67069 14 mar 08.16 avatar_db.json
>-rw-r--r-- 1 ktrls ktrls 108244 15 mar 11.31 mailboxes_db.json
>-rw-r--r-- 1 ktrls ktrls    505 15 mar 11.28 settings_db.json
>drwxr-xr-x 2 ktrls ktrls   4096 28 feb 00.58 user_dictionaries

this is the size of the dir:

> [ktrls@pohl ~]$ du -sh ~/.local/share/wmail/
> 1,6M	/home/ktrls/.local/share/wmail/

Thanks you guys for your quick answers

**Comment by Thomas101** [comment]
Thanks for that. Nothing looks suspicious there. I wonder if this is gmail writing to localstorage or something. It would be good to know if you see the high disk usage with fewer accounts. If you quit wmail and move the files out of `/.local/share/wmail/` somewhere safe (so you can restore later), restart and try with no accounts do you see high disk usage? Then if you add one do you see the same?

**Comment by ktrls** [comment]
I'm going to try and I'll tell you

**Comment by Thomas101** [comment]
Thanks :)

**Comment by ktrls** [comment]
after almost 6 hours with wmail open without accounts all works fine.

Now I'll try to add an account, on monday another, and so on.

I'll update this post when something'll go wrong

**Comment by MggMuggins** [comment]
Well, you've probably added them all at this point. I'm sure most on this thread know what I'm about to say, but I'll state it flat out, for the benefit of others reading this.
Each account uses a webview instance (am I right, or am I right?) that takes up a certain amount of RAM for itself, and then as much RAM as it takes to load the page that it's told to do (which in this case is a Gmail inbox, which is no small webpage). Each account has it's own instance of webview so that google will let you be logged in as multiple accounts at once. This isn't a huge problem when you only have three accounts, although it still is a lot of RAM. The problem gets worse as you get more accounts, to infinitum. The problem here is that we've got a lot of repeated information stored in different places in RAM, that maybe don't need to be. The question is how much of this information can you possibly condense to reduce RAM usage by a lot. My feeling is that there should be some way to store most of the code for the Gmail inboxes someplace that doesn't need to be repeated in RAM three or four or 8 times over. Anyone have any insight?

**Comment by ktrls** [comment]
Hi, I'm sorry for my silence.
At the end I have installed again the SO and have no more problems.
Maybe it was something in Electron or other JS library installed by some other program that make strange use of resorces.
If you want I can close this issue. On the other hand I don't if it can be useful to investigate.

**Comment by MggMuggins** [comment]
I'd really like to see RAM usage reduced, because I really like this app. Just how to go about that...

**Comment by jeroen7s** [comment]
i agree, it's a great app and the inbox interface is increadibly usefull, but using over 2Gb of ram for my 8 accounts, it's just too much, since i also have pycharm, opera, multiple terminals, a password manager, spotify, slack and multiple other applications open, i moved back to thunderbird because of this,which barely uses 250mb of ram with 8 accounts

**Comment by MggMuggins** [comment]
What OS are you using? I think you could benefit from LXDE...
